### DISCUSSION TITLE
Event for after scene transition completes

### DISCUSSION THREAD

**Discussion by astanb** [grahammendick/navigation#606] [Q&A] [answered]
Is there an event/hook that can be used for after a scene transition completes? Use case is a heavy render that makes the transition animation stutter, so I'd like to delay that rendering until I know the transition is complete. I'd prefer not to have to hack it with timeouts if possible.

I'm aware of `useNavigated` and `useUnloaded` but they fire instantly and don't take into account the length of the transition.

**Comment by grahammendick** [comment, marked as answer]
A heavy render shouldn't make the transition stutter. The animation is handled natively by Android/iOS - not by React or javascript. But a heavy render will prevent the animation from starting. So make sure your very first render is fast then all should be good.

**Reply by grahammendick** [reply]
That's the same behaviour as a 'real' native app, one built without react native. Android and iOS wait for the first paint before starting the animation.
